queries.sql

```diff
@@ -29,3 +29,86 @@ WHERE name <> 'Gabumon';
 SELECT *
 FROM animals
 WHERE weight_kg >= 10.4 AND weight_kg <= 17.3;
+
+------------------------------------ROLLBACK the changes----------------------------------------------
+
+BEGIN;
+
+UPDATE animals
+SET species = 'unspecified';
+
+SELECT * FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals;
+
+-----------------------------------Update the species and COMMIT--------------------------------------
+
+BEGIN; 
+
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+
+SELECT * FROM animals WHERE name LIKE '%mon';
+
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+
+SELECT * FROM animals WHERE species = 'pokemon';
+
+COMMIT;
+
+SELECT * FROM animals;
+
+
+
+-------------------------------------DELETE and ROLLBACK-------------------------------------------
+
+BEGIN; 
+
+DELETE FROM animals;
+
+ROLLBACK;
+
+SELECT * FROM animals
+
+--------------------------------------SAVEPOINT-------------------------------------------------
+
+BEGIN;
+
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT my_savepoint;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+
+ROLLBACK TO my_savepoint;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
+COMMIT;
+
+
+--------------------------------------Queries-------------------------------------------------
+
+SELECT COUNT(*) AS animal_count FROM animals;
+
+SELECT COUNT(*) AS no_escape_count FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) AS average_weight FROM animals;
+
+SELECT name, escape_attempts
+FROM animals
+WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
```

Comment:
Hello,

You've made a good attempt with your SQL submission! 🌟 It’s clear you have been paying attention to the syntax and structure of SQL queries. However, there is a mismatch between the provided solution and the project requirements.

Your Code:
```sql
SELECT name, escape_attempts
FROM animals
WHERE escape_attempts = (SELECT MAX(escape_attempts) FROM animals);
```

This code selects the `name` and `escape_attempts` for the animal(s) with the maximum number of escape attempts, which is not aligned with the requirements.

### Project Requirements 📘
You are required to find out who escapes the most, neutered, or not neutered animals. This involves grouping by the `neutered` column and either using `AVG(escape_attempts)` to find the average escape attempts or `MAX(escape_attempts)` to find the maximum escape attempts for each group (neutered and not neutered).

### Suggested Code 🛠️
Here are the correct SQL statements based on the project requirements:

Option 1 (Using `AVG`):
```sql
SELECT neutered, AVG(escape_attempts) FROM animals
GROUP BY neutered;
```

Option 2 (Using `MAX`):
```sql
SELECT neutered, MAX(escape_attempts) FROM animals
GROUP BY neutered;
```

### Conclusion 🌟

It seems like there was a misunderstanding of the project requirements. Please revise your query to reflect the needed grouping by `neutered` status and the use of either `AVG` or `MAX` functions to fulfill the project’s objective correctly. You’re making good progress, and with a little adjustment, you’ll meet the requirements perfectly. Keep up the hard work and happy coding! 👏

Reply:
done

Reply:
That's great 👍

Reply:
Hello @topeogunleye 
Still waiting for reveiw and approval. Dunno if you could help out.
Thanks